pkg/nsx/services/vpc/builder.go

```diff
@@ -28,16 +28,20 @@ func generateIPBlockKey(block model.IpAddressBlock) string {
 }
 
 func buildNSXVPC(obj *v1alpha1.NetworkInfo, nsObj *v1.Namespace, nc common.VPCNetworkConfigInfo, cluster string,
-	nsxVPC *model.Vpc, useAVILB bool) (*model.Vpc,
+	nsxVPC *model.Vpc, useAVILB bool, lbProviderChanged bool) (*model.Vpc,
 	error) {
 	vpc := &model.Vpc{}
 	if nsxVPC != nil {
 		// for upgrade case, only check public/private ip block size changing
-		if !IsVPCChanged(nc, nsxVPC) {
+		if !IsVPCChanged(nc, nsxVPC) && !lbProviderChanged {
```

Comment:
I think you should put lbProviderChanged logic in IsVPCChanged, otherwise IsVPCChanged should rename to IsVPCPrivateCIDRChanged

Reply:
vpc doesn't include lb provider, LBS created with VPC is just due to using HAPI.  without lb Provider, VPC could also be created.

Reply:
Add TODO to put lbProviderChanged into IsVPCChanged, and move IsVPCChanged out of buildNSXVPC